### PR TITLE
[AME-290] Update GTM for Pixel Metadata fields

### PR DESCRIPTION
**Pull request by elmiranda**
REDO of https://github.com/reddit/reddit-gtm-template/pull/4, from a proper reddit/reddit-gtm-template branch.

Merging into ame-200, since those changes should go in first.

---

https://reddit.atlassian.net/browse/AME-290

Added the information to the `track` param based [on how it's being retrieved in the Pixel](https://github.snooguts.net/reddit/reddit-conversion-pixel-js/pull/70/files#diff-5936bdadb9fdd3aebea680f6e1e19a3aR105).

Also included conditional display, based on the type of event being reported (c5fc10a):

Non-revenue event:
image

Purchase event:
image

Sign Up event:
image

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -96,6 +96,85 @@ ___TEMPLATE_PARAMETERS___
...
+            "name": "name",
+            "type": "SELECT",
+            "selectItems": [
+              {
+                "value": "itemCount",
+                "displayValue": "Item Count"
+              },
+              {
+                "value": "value",
+                "displayValue": "Event Value"
```
Comment (woodchuckgym) [inline template.tpl:122]: Maybe "Transaction Value" to match "Transaction ID"?

More generally -- even if you keep the name -- what about adding a parenthetical expression explaining they should use the smallest subdivision of the currency here?  I know detailed instructions come separately, but as a remember-the-human sorta thing, IMHO it's nice to have a friendly reminder that this field is not the currency base unit.  Maybe "Transaction value (pennies, paise, cents, satoshis ...)"?

File: template.tpl [hunk 2]
```diff
@@ -630,6 +712,41 @@ scenarios:
     // Verify that the tag finished successfully.
     assertApi('makeTableMap').wasCalledWith(mockData.advancedMatchingParams, 'name', 'value');
     assertApi('gtmOnSuccess').wasCalled();
+- name: Test Event Metadata
+  code: |-
+    mockData = {
+      id: "t2_potato",
```
Comment (woodchuckgym) [inline template.tpl:718]: 🥔  😂

**Review by woodchuckgym — CHANGES_REQUESTED** [review]

**Review by woodchuckgym — CHANGES_REQUESTED** [review]
Oh, can you update pixel-screenshot.png too?

**Review by monicating — APPROVED** [review]

**Review by dmaii — APPROVED** [review]

File: template.tpl (hunk 1 above)
Comment (elmiranda) [reply template.tpl:122]: Definitely for the first. For the second, I really wanted to give a little tooltip, but you can't put menu-item-level tooltips. What about `Transaction Value (in smallest subunit of currency)` versus listing out some options?

(TIL USD denomination starts at $1 and not $0.01)

**Review by woodchuckgym — CHANGES_REQUESTED** [review]
Looks good!  TY.

But the pixel-screenshot.png was only deleted and not replaced with an updated image.  Could you create a new screenshot?

**Review by eseosaasiruwa — APPROVED** [review]

**Comment by elmiranda** [comment]
> Looks good! TY.
> 
> But the pixel-screenshot.png was only deleted and not replaced with an updated image. Could you create a new screenshot?

It was showing for me on the repository if you clicked into the branch; I think the image size was just smaller? Either way, I updated it and can confirm I see the new screenshot locally, it's not deleted on the PR, and when viewing the repo on the branch, it's updated

**Review by woodchuckgym — APPROVED** [review]
Aaaaah, sorry.  Just -38 bytes.  TY!

**Pull request merged**